File: EduLink.Domain/Entities/States/CompletadaState.cs
using EduLink.Domain.Enums;

namespace EduLink.Domain.Entities.States;

public class CompletadaState : ReservaState
{
    public override EstadoReserva Nombre => EstadoReserva.Completada;

    // En estado "Completado", generalmente no se permiten más transiciones

    public override void Cancelar(Reserva reserva, DateTime ahora)
    {
        throw new InvalidOperationException("No se puede cancelar una reserva ya completada.");
    }

    public override void Completar(Reserva reserva)
    {
        // Ya está completada ea
    }

    public override void Confirmar(Reserva reserva)
    {
        // Ya fue confirmada antes de completarse ujum
    }
}

File: app/domain/entities/states/completada_state.rb
require_relative './reserva_state'
require_relative '../../enums/estado_reserva' # Corrected path

module Entities
  module States
    class CompletadaState < ReservaState
      def nombre
        Enums::EstadoReserva::COMPLETADA
      end

      # En estado "Completado", generalmente no se permiten más transiciones

      def cancelar(reserva, ahora)
        raise StandardError, 'No se puede cancelar una reserva ya completada.'
      end

      def completar(reserva)
        # Ya está completada ea
      end

      def confirmar(reserva)
        # Ya fue confirmada antes de completarse ujum
      end
    end
  end
end
